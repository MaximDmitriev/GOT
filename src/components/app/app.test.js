import React from 'react';
import App from "./app";
import {shallow} from "enzyme";

describe("Testing <App />", () => {
    const app = shallow(<App />);

    it("App have rendered", () => {
        expect(app).toMatchSnapshot();
    });

    it("app state 'showChar' is false", () => {
        expect(app.state().showChar).toBeFalse();
    });
    
    it("app state 'error' is false", () => {
        expect(app.state().error).toBeFalse();
    });
});