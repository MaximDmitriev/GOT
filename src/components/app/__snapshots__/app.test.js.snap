// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App /> App have rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef>
        <Container
          tag="div"
        >
          <Header />
        </Container>
        <Container
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              lg={
                Object {
                  "offset": 0,
                  "size": 5,
                }
              }
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            />
          </Row>
          <Row
            tag="div"
          >
            <Button
              className="float-left"
              color="secondary"
              onClick={[Function]}
              style={
                Object {
                  "margin": "15px",
                }
              }
              tag="button"
            >
              Show character
            </Button>
          </Row>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/characters/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/houses/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/books/"
            />
            <Route
              exact={true}
              path="/books/:id"
              render={[Function]}
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </Container>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Container
            tag="div"
          >
            <Header />
          </Container>,
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 0,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              />
            </Row>
            <Row
              tag="div"
            >
              <Button
                className="float-left"
                color="secondary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": "15px",
                  }
                }
                tag="button"
              >
                Show character
              </Button>
            </Row>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/characters/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/houses/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/books/"
              />
              <Route
                exact={true}
                path="/books/:id"
                render={[Function]}
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Header />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                />
              </Row>,
              <Row
                tag="div"
              >
                <Button
                  className="float-left"
                  color="secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "15px",
                    }
                  }
                  tag="button"
                >
                  Show character
                </Button>
              </Row>,
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/characters/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/houses/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/books/"
                />
                <Route
                  exact={true}
                  path="/books/:id"
                  render={[Function]}
                />
                <Route
                  component={[Function]}
                />
              </Switch>,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": null,
                  "lg": Object {
                    "offset": 0,
                    "size": 5,
                  },
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Button
                  className="float-left"
                  color="secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "15px",
                    }
                  }
                  tag="button"
                >
                  Show character
                </Button>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Show character",
                  "className": "float-left",
                  "color": "secondary",
                  "onClick": [Function],
                  "style": Object {
                    "margin": "15px",
                  },
                  "tag": "button",
                },
                "ref": null,
                "rendered": "Show character",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/characters/"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/houses/"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/books/"
                  />,
                  <Route
                    exact={true}
                    path="/books/:id"
                    render={[Function]}
                  />,
                  <Route
                    component={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/characters/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/houses/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/books/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "exact": true,
                    "path": "/books/:id",
                    "render": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": true,
          "rules": Array [
            "
    overflow-x: hidden;
    background: url(",
            "got.jpeg",
            ") center center no-repeat;
    background-size: cover;
    font-size: 16px;
    height: 100vh;
    button{
        box-shadow: none !important;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <Container
            tag="div"
          >
            <Header />
          </Container>
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 0,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              />
            </Row>
            <Row
              tag="div"
            >
              <Button
                className="float-left"
                color="secondary"
                onClick={[Function]}
                style={
                  Object {
                    "margin": "15px",
                  }
                }
                tag="button"
              >
                Show character
              </Button>
            </Row>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/characters/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/houses/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/books/"
              />
              <Route
                exact={true}
                path="/books/:id"
                render={[Function]}
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </Container>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Container
              tag="div"
            >
              <Header />
            </Container>,
            <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                />
              </Row>
              <Row
                tag="div"
              >
                <Button
                  className="float-left"
                  color="secondary"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "15px",
                    }
                  }
                  tag="button"
                >
                  Show character
                </Button>
              </Row>
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/characters/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/houses/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/books/"
                />
                <Route
                  exact={true}
                  path="/books/:id"
                  render={[Function]}
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </Container>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Header />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Row
                  tag="div"
                >
                  <Col
                    lg={
                      Object {
                        "offset": 0,
                        "size": 5,
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  />
                </Row>,
                <Row
                  tag="div"
                >
                  <Button
                    className="float-left"
                    color="secondary"
                    onClick={[Function]}
                    style={
                      Object {
                        "margin": "15px",
                      }
                    }
                    tag="button"
                  >
                    Show character
                  </Button>
                </Row>,
                <Switch>
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/characters/"
                  />
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/houses/"
                  />
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/books/"
                  />
                  <Route
                    exact={true}
                    path="/books/:id"
                    render={[Function]}
                  />
                  <Route
                    component={[Function]}
                  />
                </Switch>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Col
                    lg={
                      Object {
                        "offset": 0,
                        "size": 5,
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  />,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": null,
                    "lg": Object {
                      "offset": 0,
                      "size": 5,
                    },
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Button
                    className="float-left"
                    color="secondary"
                    onClick={[Function]}
                    style={
                      Object {
                        "margin": "15px",
                      }
                    }
                    tag="button"
                  >
                    Show character
                  </Button>,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Show character",
                    "className": "float-left",
                    "color": "secondary",
                    "onClick": [Function],
                    "style": Object {
                      "margin": "15px",
                    },
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": "Show character",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/"
                    />,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/characters/"
                    />,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/houses/"
                    />,
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/books/"
                    />,
                    <Route
                      exact={true}
                      path="/books/:id"
                      render={[Function]}
                    />,
                    <Route
                      component={[Function]}
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/characters/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/houses/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/books/",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "exact": true,
                      "path": "/books/:id",
                      "render": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": true,
            "rules": Array [
              "
    overflow-x: hidden;
    background: url(",
              "got.jpeg",
              ") center center no-repeat;
    background-size: cover;
    font-size: 16px;
    height: 100vh;
    button{
        box-shadow: none !important;
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
